lib/spotlight/dor/indexer.rb

```diff
@@ -182,6 +182,52 @@ def add_folder_name(sdb, solr_doc)
       end
     end
 
+    concerning :FullTextIndexing do
+      included do
+        before_index :add_object_full_text
+      end
+
+      # search for configured full text files, and if found, add them to the full text (whole document) solr field
+      def add_object_full_text(sdb, solr_doc)
+        object_level_full_text_urls(sdb).each do |file_url|
+          # append content from each file to the field, creating field if it doesn't exist yet
+          # ruby note: the construct below allows us to append a string to a variable that starts out as nil
+          # because .to_s of nil is empty string
+          solr_doc['full_text_tesim'] = solr_doc['full_text_tesim'].to_s + get_file_content(file_url)
+        end
+      end
+
+      # go grab the supplied file url, grab the file, encode and return
+      # TODO: thse should also be able to also deal with .rtf and .xml files
+      def get_file_content(file_url)
+        response = Net::HTTP.get_response(URI.parse(file_url))
```

Comment:
I think spotlight is bringing in Faraday already. It might do some nicer error handling or timeouts for us. Something to think about, at least.